### PR TITLE
Recommended MySql version bumbed from 5.5.3 to 5.7.14

### PR DESCRIPTION
**Pull request by escopecz**
**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | N
| Automated tests included? | N
| Related user documentation PR URL | /
| Related developer documentation PR URL | /
| Issues addressed (#s or URLs) | https://acquia.slack.com/archives/CPVGZE516/p1578392663001500
| BC breaks? | Yes, but it's only a recommendation, not hard limit.
| Deprecations? | N

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

I asked the community in Slack if we can bump the MySql version and there was no one against. It will let us use more advanced features like https://github.com/mautic/mautic/pull/7094.

#### Steps to test this PR:
1. Just a readme change. Nothing to test. Just review.

### CODE REVIEW

File: UPGRADE-3.0.md [hunk 1]
```diff
@@ -13,6 +13,7 @@
 # Backwards compatibility breaking changes
 
 *   Minimal PHP version was increased from v5.6.19 to v7.2.0.
+*   Minimal MySql version was increased from v5.5.3 to v5.7.14
```
Comment (anton-vlasenko) [inline UPGRADE-3.0.md]:
```suggestion
*   Minimal MySQL version was increased from v5.5.3 to v5.7.14
```
This is really a nitpick. But I think we should use the same spelling in all of the documents (in README.md we spell it as `MySQL`).

**Review by anton-vlasenko — CHANGES_REQUESTED** [review]

**Review by anton-vlasenko — APPROVED** [review]
@escopecz Could you please merge this PR?

**Comment by digitall-it** [comment]
Just wanted to let you know that I cannot upgrade from v2.x to v3.0.1 because I'm stuck with MySQL v5.5.65.

**Comment by escopecz** [comment]
I'm sorry about that. But consider changing your server provider if you are forced to use unsupported software versions. According to https://fromdual.com/support-for-mysql-from-oracle the extended support for 5.5 ended in Dec 2018. This is a security risk.

**Comment by digitall-it** [comment]
I understand, but on CentOS 7, a distro still widely used, upgrading from MariaDB 5.5 to 10.x (a drop-in replacement for MySQL 5.7.x) is a major task, which many didn't discover they had to do until now that they have to upgrade Mautic to 3.x and are stuck with the older version for now.

**Pull request merged**